framework/src/headers/ip/v6/ext.rs

```diff
@@ -0,0 +1,62 @@
+use headers::ip::v6::{Ipv6VarHeader, NextHeader};
+use headers::EndOffset;
+use num::FromPrimitive;
+use std::marker::PhantomData;
+
+#[derive(Default)]
+#[repr(C, packed)]
+// All extension headers share the first two bytes, which are the next_header
+// field and the header length. We can use this struct as the first section of a
+// more specific header, or as a way to skip extension headers in the processing
+// pipeline that we don't care about.
+pub struct Ipv6ExtHeader<T>
+where
+    T: Ipv6VarHeader,
+{
+    pub next_header: u8,
+    pub hdr_ext_len: u8,
+    _parent: PhantomData<T>,
+}
+
+// Generic extension headers have the next_header field.
+impl<T> Ipv6VarHeader for Ipv6ExtHeader<T>
+where
+    T: Ipv6VarHeader,
+{
+    fn next_header(&self) -> Option<NextHeader> {
+        FromPrimitive::from_u8(self.next_header)
+    }
+}
+
+impl<T> EndOffset for Ipv6ExtHeader<T>
+where
+    T: Ipv6VarHeader,
+{
+    type PreviousHeader = T;
+
+    #[inline]
+    fn offset(&self) -> usize {
+        // Hdr Ext Len: 8-bit unsigned integer, is the length of the extension
+        // header in 8-octet units, not including the first 8 octets.
+        (self.hdr_ext_len as usize) * 8 + 8
```

Comment:
:+1: Had a long comment written up but it's just that "8-octet units" is too meta for me. This is right.